recipe/install_python.sh

```diff
@@ -1,4 +1,6 @@
 python -c "import numpy"
+echo `which python`
+echo "$PYTHON"
```

Comment:
Right, the recipe uses `outputs`, so possibly you run in some kind of variant of https://github.com/conda/conda-build/issues/3993 , i.e., I wouldn't be surprised if `$PYTHON` is not set correctly.

Reply:
But even then,
> (But overall, it is really strange that it finds the correct interpreter from the host environment but then uses the system's one for the NumPy check; no idea why that happens.)

would still be strange.